samples/src/main/java/terminal/TerminalClient.java

```diff
@@ -17,100 +17,119 @@
+                .getTerminalId();
+        LOG.info("Selected terminal: {}", exTerminalId);
+
+        return response;
+    }
+
+    public PayResponse sendPayment(String amountStr, boolean showTips, PaymentMethodType paymentMethodType) {
         LOG.info("Sending payment for amount: {} to terminal: {}", amountStr, exTerminalId);
         BigDecimal amount = new BigDecimal(amountStr);
-        CompletableFuture<PayResponse> response = client.sendPayment(exTerminalId, amount, showTips, paymentMethodType, orderId -> {
-            LOG.info("onPending: orderId={}", orderId);
-            // optionally cancel payment after delay
-            Executor delayed = CompletableFuture.delayedExecutor(30L, TimeUnit.SECONDS);
-
-            CompletableFuture.supplyAsync(() -> {
-                LOG.info("Cancelling payment: {} = {}", amount, orderId);
-                return cancelPayment(amountStr, orderId);
-            }, delayed);
-        });
-        response.thenAccept(res -> LOG.info("Sent Payment: {}, {}", amount, dump(res)));
-        return response.get(timeout, TimeUnit.MINUTES);
+
+        PayRequest payRequest = PayRequest.newBuilder()
+                .setStoreId(storeId.toString())
+                .setAmount(amount.toPlainString())
+                .setTerminalId(exTerminalId)
+                .setShowTips(showTips)
+                .setPaymentMethod(PaymentMethod.newBuilder().setPaymentMethodType(paymentMethodType).build())
+                .build();
+
+        PayResponse response = paymentClient.sendTerminalPayment(payRequest);
+
+        LOG.info("Sent Payment: {}, {}", amount, dump(response));
+        return response;
     }
 
-    public CompletableFuture<PaymentStatus> cancelPayment(String amountStr, String orderId) {
+    public PaymentStatus cancelPayment(String amountStr, String orderId) {
         LOG.info("Cancel payment: {} with amount: {}", orderId, amountStr);
         BigDecimal amount = new BigDecimal(amountStr);
 
-        CompletableFuture<PaymentStatus> response = client.cancelPayment(amount, exTerminalId, orderId);
+        CancelRequest cancelRequest = CancelRequest.newBuilder()
+                .setStoreId(storeId.toString())
+                .setAmount(amount.toPlainString())
+                .setTerminalId(exTerminalId)
+                .setOrderId(orderId)
+                .build();
 
-        response.thenAccept(res -> LOG.info("Cancelled Payment: {} = {}", orderId, res));
+        PaymentStatus response = paymentClient.cancelPayment(cancelRequest);
+
+        LOG.info("Cancelled Payment: {} = {}", orderId, response);
 
         return response;
     }
 
-    public RefundResponse requestRefund(String amountStr, String orderId) throws ExecutionException, InterruptedException, TimeoutException {
-        Executor delayed = CompletableFuture.delayedExecutor(5L, TimeUnit.SECONDS);
+    public RefundResponse requestRefund(String amountStr, String orderId) throws InterruptedException {
+        Thread.sleep(5000);
```

Comment:
This 5 second delay should be within the _calling code_ not within this sample method, IMO